anni-provider/src/providers/file.rs

```diff
@@ -128,51 +149,29 @@ impl AnniProvider for FileBackend {
     ) -> Result<AudioResourceReader, ProviderError> {
         let path = self.get_disc(album_id, disc_id)?;
         let mut dir = read_dir(path).await?;
-        while let Some(entry) = dir.next_entry().await? {
-            let filename = entry.file_name();
-            if filename
-                .to_string_lossy()
-                .starts_with::<&str>(format!("{:02}.", track_id).as_ref())
-            {
-                let path = entry.path();
-                let mut file = File::open(&path).await?;
-                let metadata = file.metadata().await?;
-                let file_size = metadata.len();
-
-                // limit in range
-                file.seek(SeekFrom::Start(range.start)).await?;
-                let file = file.take(range.length_limit(file_size));
-
-                // calculate audio duration only if it flac header is in range
-                let (duration, reader): (u64, ResourceReader) = if range.contains_flac_header() {
-                    let (info, reader) = crate::utils::read_header(file).await?;
-                    (info.total_samples / info.sample_rate as u64, reader)
-                } else {
-                    (0, Box::pin(file))
-                };
-
-                return Ok(AudioResourceReader {
-                    info: AudioInfo {
-                        extension: path.extension().map(|s| s.to_string_lossy().to_string()).unwrap_or_default(),
-                        size: file_size as usize,
-                        duration,
-                    },
-                    range: if range.is_full() {
-                        range
-                    } else {
-                        range.end_with(file_size)
-                    },
-                    reader,
-                });
+        let file = loop {
+            match dir.next_entry().await? {
+                Some(entry)
+                    if entry
+                        .file_name()
+                        .to_string_lossy()
+                        .starts_with(&format!("{:02}.", track_id)) =>
+                {
+                    break File::open(entry.path()).await?
+                }
+                None => return Err(ProviderError::FileNotFound),
+                _ => {}
```

Comment:
```suggestion
```

unreachable?

Reply:
The `Some` branch has a match guard to match `track id`.
The branch is reached when `next_entry()` exists but is not the intended track.